Automattic/woocommerce-payments
# Transactions: Use server based export for both sync download on browser and async email export

Alternative approach for  #9969

#### Changes proposed in this Pull Request
In this PR, the UX for browser download remains the same, but the backend implementation is changed so that the server export is used for generating the CSV which is made available immediately. This approach does NOT change the UX for merchant (i.e. download is available immediately for single page export), but it solves the overhead and issues of maintaining multiple implementations that do not match.

Single page downloads are available immediately, multiple page downloads (i.e. number of rows to be exported > Page size ) the download is emailed.

The corresponding server changes are on https://github.com/Automattic/transact-platform-server/pull/7077.

**NOTE:**
- We may need to make data on screen consistent with the export data, or message appropriately. e.g. custom order number is available on screen but not from server.

#### Testing instructions

- If the number of transactions to be downloaded are less than page size, file is available for download immediately. 
- If there are multiple pages, the download is emailed to the admin email as before.
- NOTE: UX is unchanged.

Test 1:
- There are < 25 rows of transactions listed and page size is 25
- Download should be immediate

Test 2:
- The number of transactions filtered and displayed on listing page is between 25 and 100
- When page size is 25 (default), the download is emailed.
- Change the page size to 100, so that there is only one page of transactions, the download is available immediately, but NOT emailed

Test 3:
- There are > 100 transactions listed
- The download is emailed.

-------------------


**Post merge**
